File: C++/204. Count Primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n==0 || n==1)
            return 0;
        vector<bool> is_prime(n, true);
        int ans = 0;
        is_prime[0] = is_prime[1] = false;
        for(int i = 2;i<n;i++)
        {
            if(is_prime[i])
            {
                ans++;
                for(int j=2*i;j<n;j+=i)
                    is_prime[j] = false;
            }
        }
        return ans;
    }
};

File: Python/204. Count Primes.py
class Solution:
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0 or n == 1:
            return 0
        is_prime = [True] * n
        ans = 0
        is_prime[0] = is_prime[1] = False
        for i in range(2, n):
            if is_prime[i]:
                ans += 1
                for j in range(2*i, n, i):
                    is_prime[j] = False
        return ans
        